src/templates/java/JavaTemplateProvider.ts

```diff
@@ -22,42 +22,41 @@ interface IRawJavaTemplates {
     templates: object[];
 }
 
-/**
- * Java templates largely follow the same formatting as script templates, but they come from maven
- */
+const repositoryUrl: string = "https://repo1.maven.org/maven2/com/microsoft/azure/azure-functions-maven-plugin/maven-metadata.xml";
+const templatesUrl: string = "https://aka.ms/java-function-templates";
```

Comment:
Please work with the Functions team to get these templates added to the main templates repo and templates feed

- Repo: https://github.com/Azure/azure-functions-templates
- Feed: https://functionscdn.azureedge.net/public/cli-feed-v4.json

I don't think using a GitHub url directly is reliable enough for use in the released version of our extension.

Reply:
@ejizba Thanks for your feedback. We are now syncing with Function team to add Java to templates repo, but there seems to be some history issue there, I'll update here once there are updates.

Besides, now `JavaTemplateProvider` depends on functions maven plugin, which only works for maven project. Get templates from git repo could help users get the latest java templates for all projects, besides, we could fall back to backup template shipped with extension once there are issues with Github connection.  Or we could keep origin java template provider but just update the java function creation step, which will keep current behavior for maven and use shipped template for other java projects(like gradle)?

Reply:
I don't like using a link to GitHub. The cli-feed has an established process for moving templates to a "staging" environment first (that's what the "prerelease" tags are for) and we run unit tests against that environment to make sure everything looks good before it moves to "production". A direct link to GitHub does not have any process like that, so it's much more possible a change can break the released version of our extension.

I'm okay with using the backup templates however you want to do that. That option is much less prone to bugs since everything is just shipped with the extension and there's no "external" points of integration over the internet. Obviously the main downside is that you have to wait for the next release of the extension to update the templates

Reply:
Yes, I agree that GitHub link is not a stable solution for function template, I've revert the changes for Java template provider but just update the FunctionCreationStep, will update the template provider once we add java templates to template repository.